src/app/pages/orders/components/orders/orders.component.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  HostBinding,
+  OnInit,
+  inject,
+  DestroyRef,
+} from '@angular/core';
+import { IOrder, OrdersFacadeService, ProductFacadeService } from '@shared-module';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+@Component({
+  selector: 'app-orders',
+  templateUrl: './orders.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OrdersComponent implements OnInit {
+  orders: IOrder[];
+
+  private readonly destroyRef = inject(DestroyRef);
```

Comment:
It was my mistake to use the destroyRef as a class field.
You remember that we were discussing the inject disadvantage that we need to declare many class fields for services etc. But DestroyRef is also a service, like ChangeDetectorRef, so we can use it directly in a class constructor in all places of the project.